return nil

The following program `sample.go` triggers a panic:

```go
package main        
                
import "fmt"
    
func f() (host, port string, err error) {
    return "", "", nil
}
    
func main() {
    h, p, err := f()
    fmt.Println(h, p, err)
}
```

Expected result:
```console
$ go run ./sample.go
  <nil>
```

Got:
```console
$ yaegi ./sample.go
panic: runtime error: index out of range [2] with length 2 [recovered]
        panic: _test/nil0.go:4:2: CFG post-order panic: runtime error: index out of range [2] with length 2

goroutine 1 [running]:
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2.1(0xc00018b100)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:366 +0xba
panic(0xd775e0, 0xc000036240)
        /usr/lib/go/src/runtime/panic.go:679 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).cfg.func2(0xc00018b100)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:1071 +0xb5db
github.com/containous/yaegi/interp.(*node).Walk(0xc00018b100, 0xc0001bbce8, 0xc0001bbd20)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:171 +0xaf
github.com/containous/yaegi/interp.(*node).Walk(0xc00018b000, 0xc0001bbce8, 0xc0001bbd20)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc00018a300, 0xc0001bbce8, 0xc0001bbd20)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*node).Walk(0xc00018a100, 0xc0001bbce8, 0xc0001bbd20)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:168 +0x70
github.com/containous/yaegi/interp.(*Interpreter).cfg(0xc000260b00, 0xc00018a100, 0xc0000341b0, 0x4, 0x0, 0x0, 0x0)
        /home/marc/go/src/github.com/containous/yaegi/interp/cfg.go:48 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000260b00, 0xc000272000, 0x85, 0x285, 0xc000272000, 0x85, 0x0, 0x1)
        /home/marc/go/src/github.com/containous/yaegi/interp/interp.go:322 +0x1ee
main.main()
        /home/marc/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:82 +0x42e
```

--- a/_test/nil0.go
+++ b/_test/nil0.go
@@ -1,0 +1,15 @@
+package main
+
+import "fmt"
+
+func f() (host, port string, err error) {
+	return "", "", nil
+}
+
+func main() {
+	h, p, err := f()
+	fmt.Println(h, p, err)
+}
+
+// Output:
+//   <nil>

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1068,7 +1068,7 @@ func (interp *Interpreter) cfg(root *node) ([]*node, error) {
 					// nil: Set node value to zero of return type
 					f := sc.def
 					var typ *itype
-					if typ, err = nodeType(interp, sc, f.child[2].child[1].child[i].lastChild()); err != nil {
+					if typ, err = nodeType(interp, sc, f.child[2].child[1].fieldType(i)); err != nil {
 						return
 					}
 					if typ.cat == funcT {
@@ -1641,6 +1641,29 @@ func (n *node) isNatural() bool {
 		}
 	}
 	return false
+}
+
+// fieldType returns the nth parameter field node (type) of a fieldList node
+func (n *node) fieldType(m int) *node {
+	k := 0
+	l := len(n.child)
+	for i := 0; i < l; i++ {
+		cl := len(n.child[i].child)
+		if cl < 2 {
+			if k == m {
+				return n.child[i].lastChild()
+			}
+			k++
+			continue
+		}
+		for j := 0; j < cl-1; j++ {
+			if k == m {
+				return n.child[i].lastChild()
+			}
+			k++
+		}
+	}
+	return nil
 }
 
 // lastChild returns the last child of a node

